hazelcast/src/test/java/com/hazelcast/config/XMLConfigPreProcessorTest.java

```diff
@@ -500,6 +501,56 @@ public void testXmlDeniesDuplicateMemberAttributesConfig() throws Exception {
         configBuilder.build();
     }
 
+    @Test
+    public void testXmlVariableReplacementAsSubstring() throws Exception {
+        Properties properties = new Properties();
+        String xml = "<hazelcast>\n" +
+                "    <properties>\n" +
+                "        <property name=\"${env}-with-suffix\">local-with-suffix</property>\n" +
+                "        <property name=\"with-prefix-${env}\">with-prefix-local</property>\n" +
+                "    </properties>\n" +
+                "</hazelcast>";
+
+        properties.setProperty("env", "local");
+
+        ByteArrayInputStream bis = new ByteArrayInputStream(xml.getBytes());
+        XmlConfigBuilder configBuilder = new XmlConfigBuilder(bis);
+        configBuilder.setProperties(properties);
+        Config config = configBuilder.build();
+
+        assertEquals(config.getProperty("local-with-suffix"), "local-with-suffix");
+        assertEquals(config.getProperty("with-prefix-local"), "with-prefix-local");
+    }
+
+    @Test
+    public void testXmlImportWithVariableReplacementAsSubstring() throws Exception {
+        Properties properties = new Properties();
+        File file = createConfigFile("foo", "bar");
+        FileOutputStream os = new FileOutputStream(file);
+        String networkConfig = "<hazelcast>" +
+                "    <network>\n" +
+                "        <join>\n" +
+                "            <multicast enabled=\"false\"/>\n" +
+                "            <tcp-ip enabled=\"true\"/>\n" +
+                "        </join>\n" +
+                "    </network>\n" +
+                "</hazelcast>";
+        writeStringToStreamAndClose(os, networkConfig);
+
+        String xml = "<hazelcast>\n" +
+                "    <import resource=\"file://" + "${file}" + "\"/>\n" +
+                "</hazelcast>";
+
+        ByteArrayInputStream bis = new ByteArrayInputStream(xml.getBytes());
+        properties.setProperty("file", file.getAbsolutePath());
+        XmlConfigBuilder configBuilder = new XmlConfigBuilder(bis);
+        configBuilder.setProperties(properties);
+        Config config = configBuilder.build();
+        JoinConfig join = config.getNetworkConfig().getJoin();
+        assertFalse(join.getMulticastConfig().isEnabled());
```

Comment:
I see you are still relying on network config to determine of the substitution has worked. I really would use the same approach as you do in test testXmlVariableReplacementAsSubstring